manifests/bootstrap_rpm.pp

```diff
@@ -19,51 +16,52 @@
 
   $katello_rhsm_setup_script = 'katello-rhsm-consumer'
   $katello_rhsm_setup_script_location = "/usr/bin/${katello_rhsm_setup_script}"
+  $candlepin_cert_rpm_alias_filename = 'katello-ca-consumer-latest.noarch.rpm'
 
-  $candlepin_consumer_name = "katello-ca-consumer-${rhsm_hostname}"
-  $candlepin_consumer_summary = "Subscription-manager consumer certificate for Katello instance ${rhsm_hostname}"
-  $candlepin_consumer_description = 'Consumer certificate and post installation script that configures rhsm.'
+  $bootstrap_rpm_name = "katello-ca-consumer-${rhsm_hostname}"
 
   include trusted_ca
   trusted_ca::ca { 'katello_server-host-cert':
     source  => $katello_server_ca_cert,
     require => File[$katello_server_ca_cert],
   }
 
+  package { 'rpm-build':
+    ensure => installed,
+  }
+
   include apache
-  file { $katello_www_pub_dir:
+  file { $rpm_serve_dir:
     ensure => directory,
     owner  => 'apache',
     group  => 'apache',
     mode   => '0755',
-  } ->
+  }
+
   # Placing the CA in the pub dir for trusting by a user in their browser
-  file { "${katello_www_pub_dir}/${server_ca_name}.crt":
+  file { "${rpm_serve_dir}/${server_ca_name}.crt":
     ensure  => file,
     source  => $katello_server_ca_cert,
     owner   => 'root',
     group   => 'root',
     mode    => '0644',
     require => File[$katello_server_ca_cert],
-  } ~>
-  foreman_proxy_content::rhsm_reconfigure_script { "${katello_www_pub_dir}/${katello_rhsm_setup_script}":
-    ca_cert         => $ca_cert,
+  } ->
+  rhsm_reconfigure_script { "${rpm_serve_dir}/${katello_rhsm_setup_script}":
+    ensure          => present,
+    default_ca_cert => $ca_cert,
     server_ca_cert  => $katello_server_ca_cert,
-    rhsm_ca_dir     => $rhsm_ca_dir,
     default_ca_name => $default_ca_name,
     server_ca_name  => $server_ca_name,
     rhsm_hostname   => $rhsm_hostname,
     rhsm_port       => $rhsm_port,
-    rhsm_prefix     => $rhsm_prefix,
-  } ~>
-  certs_bootstrap_rpm { $candlepin_consumer_name:
-    dir              => $katello_www_pub_dir,
-    summary          => $candlepin_consumer_summary,
-    description      => $candlepin_consumer_description,
-    files            => ["${katello_rhsm_setup_script_location}:755=${katello_www_pub_dir}/${katello_rhsm_setup_script}"],
-    bootstrap_script => "/bin/bash ${katello_rhsm_setup_script_location}",
-    postun_script    => file('certs/postun.sh'),
-    alias            => $candlepin_cert_rpm_alias_filename,
-    subscribe        => $server_ca,
+  }
+
+  bootstrap_rpm { $bootstrap_rpm_name:
+    ensure  => present,
+    script  => "${rpm_serve_dir}/${katello_rhsm_setup_script}",
+    dest    => $rpm_serve_dir,
+    symlink => "${rpm_serve_dir}/${candlepin_cert_rpm_alias_filename}",
+    require => Rhsm_reconfigure_script["${rpm_serve_dir}/${katello_rhsm_setup_script}"],
```

Comment:
This suggests it could have an autorequire in the type:
```ruby
autorequire(:rhsm_reconfigure_script) do
  [self[:script]]
end
```

Reply:
Already does -- https://github.com/theforeman/puppet-foreman_proxy_content/pull/274/files#diff-b3f88bbcbe8556735e4d1d0ec3017af9R26

Reply:
In that case this line is redundant and can be dropped.